pkg/sql/set_var.go

```diff
@@ -182,6 +188,41 @@ func (n *setVarNode) Next(_ runParams) (bool, error) { return false, nil }
 func (n *setVarNode) Values() tree.Datums            { return nil }
 func (n *setVarNode) Close(_ context.Context)        {}
 
+func (n *resetAllNode) startExec(params runParams) error {
+	for varName, v := range varGen {
+		if v.Set == nil && v.RuntimeSet == nil && v.SetWithPlanner == nil {
+			continue
+		}
+		_, defVal := getSessionVarDefaultString(
+			varName,
+			v,
+			params.p.sessionDataMutatorIterator.sessionDataMutatorBase,
+		)
+		if err := params.p.SetSessionVar(params.ctx, varName, defVal, false /* isLocal */); err != nil {
+			return err
+		}
+	}
+	for varName := range params.SessionData().CustomOptions {
+		_, v, err := getSessionVar(varName, false /* missingOK */)
+		if err != nil {
+			return err
+		}
+		_, defVal := getSessionVarDefaultString(
+			varName,
+			v,
+			params.p.sessionDataMutatorIterator.sessionDataMutatorBase,
+		)
+		if err := params.p.SetSessionVar(params.ctx, varName, defVal, false /* isLocal */); err != nil {
```

Comment:
should we be a little more careful and only call `SET` if it is not equal the default?

Reply:
can you also add a test for SET ROLE? also, can you check current_user/session_user after? that one is a bit special

Reply:
good ideas!

Reply:
actually, don't think we want the call to only happen if it is not the default -- for example:

```
-- application_name default is ""

SET application_name = 'abc';
BEGIN;
SET LOCAL application_name = '';
RESET ALL;
COMMIT;

SHOW application_name;
-- should show ''
```

Reply:
(Y)